Source/EasyNetQ/PullingConsumerExtensions.cs

```diff
@@ -0,0 +1,177 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace EasyNetQ
+{
+    /// <summary>
+    ///     The item of the messages batch
+    /// </summary>
+    public readonly struct PullBatchItem
+    {
+        /// <summary>
+        ///     Creates PullBatchItem
+        /// </summary>
+        /// <param name="receivedInfo">The received info</param>
+        /// <param name="properties">The properties</param>
+        /// <param name="body">The body</param>
+        public PullBatchItem(MessageReceivedInfo receivedInfo, MessageProperties properties, byte[] body)
+        {
+            ReceivedInfo = receivedInfo;
+            Properties = properties;
+            Body = body;
+        }
+
+        /// <summary>
+        ///     A received info associated with the message
+        /// </summary>
+        public MessageReceivedInfo ReceivedInfo { get; }
+
+        /// <summary>
+        ///     Various message properties
+        /// </summary>
+        public MessageProperties Properties { get; }
+
+        /// <summary>
+        ///     The message body
+        /// </summary>
+        public byte[] Body { get; }
+    }
+
+    /// <summary>
+    ///     The result of a pull batch operation
+    /// </summary>
+    public readonly struct PullBatchResult
+    {
+        /// <summary>
+        ///     Creates PullBatchResult
+        /// </summary>
+        /// <param name="messages">The messages</param>
+        public PullBatchResult(IReadOnlyList<PullBatchItem> messages)
+        {
+            Messages = messages;
+        }
+
+        /// <summary>
+        ///     Messages of the batch
+        /// </summary>
+        public IReadOnlyList<PullBatchItem> Messages { get; }
+
+        /// <summary>
+        ///     Returns delivery tag of the batch
+        /// </summary>
+        /// <exception cref="InvalidOperationException"></exception>
+        public ulong DeliveryTag
+        {
+            get
+            {
+                if (Messages.Count == 0)
+                    throw new InvalidOperationException("No messages");
+
+                return Messages.Max(x => x.ReceivedInfo.DeliveryTag);
+            }
+        }
+    }
+
+    /// <summary>
+    ///     Various extensions for IPullingConsumer
+    /// </summary>
+    public static class PullingConsumerExtensions
```

Comment:
So there are extensions only for `IPullingConsumer` but not for its generic counterpart `IPullingConsumer<T>`. This is one more argument to inherit second from first.